docs/docs/metrics.md

```diff
@@ -0,0 +1,150 @@
+this is one number which influences the performance of Marathon. If you have
+a high number of groups, your performance will be lower than for a low number of
+groups. Note that each level between the slashes in your app IDs corresponds to
+a group. The app `/shop/frontend` is in the `frontend` group which is in
+the `shop` group which is in the root group.
+
+<span class="label label-default">v0.15</span>
+`service.mesosphere.marathon.task.running.count` (gauge) - The number of tasks that are
+currently running.
+
+<span class="label label-default">v0.15</span>
+`service.mesosphere.marathon.task.staged.count` (gauge) - The number of tasks that are
+currently staged. After tasks are launched, they are first in this state.
+A consistently high number of staged tasks indicates a lot of churn in Marathon
+and Mesos. Either you have many app updates/manual restarts or some of your apps
+have stability problems and are consistently automatically restarted.
+
+### Task update processing
+
+<span class="label label-default">v0.15</span>
+`service.mesosphere.marathon.core.task.update.impl.ThrottlingTaskStatusUpdateProcessor.queued` (gauge) -
+ number of queued status updates.
+
+<span class="label label-default">v0.15</span>
+`service.mesosphere.marathon.core.task.update.impl.ThrottlingTaskStatusUpdateProcessor.processing` (gauge) -
+ number of currently processing status updates.
+ 
+ <span class="label label-default">v0.15</span>
+ `service.mesosphere.marathon.core.task.update.impl.TaskStatusUpdateProcessorImpl.publishFuture` (timer) -
+ times processing status updates.
+
+### Configuration update processing
+
+<span class="label label-default">v0.15</span>
+`service.mesosphere.marathon.state.GroupManager.queued` (gauge) - number of queued app configuration
+updates. The maximum is configured with `--max_queued_root_group_updates`.
+
+<span class="label label-default">v0.15</span>
+`service.mesosphere.marathon.state.GroupManager.processing` (gauge) - number of currently processed app configuration
+updates. Since we serialize these updates, this is either 0 or 1.
+
+### Repositories
+
+Marathon stores its permanent state in "Repositories". These are the most important ones:
+
+* `GroupRepository` for app configurations and groups.
+* `TaskRepository` for the latest known task state. This is the repository with the largest
+  data churn.
+
+There are also the
+
+* AppRepository for versioned app configuration.
+* DeploymentRepository for currently running deployments.
+* TaskFailureRepository for the last failure for every application.
+* and more...
+
+For each of the repositories we have statistics about read and write requests. Substitute `*` with
+the repository of interest:
+ 
+`service.mesosphere.marathon.state.*.read-request-time.count` - the number of read requests.
+
+`service.mesosphere.marathon.state.*.read-request-time.mean` - the exponential weighted average of
```

Comment:
Not we hourselves but other metric solutions might layer on top of that. But as with the mean I don't really know how to interpret them other than "oh, that's really long", "that's probably not the problem". The problem is that we don't really report this reliably for reporting intervals in combination with technologies as datadog, statsd, graphite.

```
"service.mesosphere.marathon.state.AppRepository.write-request-time": {
      "count": 43,
      "max": 3,
      "mean": 1.1844599999513041,
      "min": 1,
      "p50": 1.0,
      "p75": 1.0,
      "p95": 2.0,
      "p98": 2.0,
      "p99": 2.0,
      "p999": 2.0,
      "stddev": 0.3878588768731083
    },
```